ci/cpu/build.sh

```diff
@@ -78,7 +78,8 @@ conda config --set ssl_verify False
 
 # TODO: Move boa install to gpuci/rapidsai
 gpuci_mamba_retry install boa
-
+gpuci_conda_retry remove --force pyarrow arrow-cpp
+gpuci_mamba_retry install -y "pyarrow=10.0.1" "arrow-cpp=10.0.1"
```

Comment:
TODO: Need to remove these after https://github.com/rapidsai/integration/pull/577 is merged.